Tests/MapboxMapsTests/Snapshot/MapboxMapsSnapshotTests.swift

```diff
@@ -1,10 +1,150 @@
 import XCTest
+import MapboxMaps
+import MapboxCoreMaps
+import Turf
+import Foundation
+import UIKit
 #if canImport(MapboxMaps)
 @testable import MapboxMaps
 #else
 @testable import MapboxMapsSnapshot
 #endif
-
 //swiftlint:disable explicit_top_level_acl explicit_acl
+
 class MapboxMapsSnapshotTests: XCTestCase {
+
+    var newAttachment: XCTAttachment!
+    var resourceOptions: ResourceOptions!
+
+    // Create snapshot options
+    private func snapshotterOptions() throws -> MapSnapshotOptions {
+        let accessToken = try mapboxAccessToken()
+        resourceOptions = ResourceOptions(accessToken: accessToken)
+        return MapSnapshotOptions(size: CGSize(width: 300, height: 300), pixelRatio: 4, resourceOptions: resourceOptions)
+    }
+
+    // Testing creating the snapshot
+    func testSnapshotCancellation() throws {
+        weak var weakSnapshotter: Snapshotter?
+        let options = try snapshotterOptions()
+        let expectation = self.expectation(description: "snapshot")
+         autoreleasepool {
+            let snapshotter = Snapshotter(options: options)
+            weakSnapshotter = snapshotter
+            weakSnapshotter?.setCamera(to: CameraOptions(center: CLLocationCoordinate2D(latitude: 38.9180379, longitude: -77.0600235), zoom: 5))
+            weakSnapshotter?.style.uri = .light
+            weakSnapshotter?.start(overlayHandler: nil) { (result) in
+                expectation.fulfill()
+                XCTAssertNotNil(result)
+                XCTAssertNil(weakSnapshotter)
+            }
+        }
+        XCTAssertNil(weakSnapshotter)
+        wait(for: [expectation], timeout: 10)
+    }
+
+    func TODO_testCapturingSnapshotterInSnapshotCompletion() throws {
```

Comment:
These tests marked with `TODO_` are temporarily disabled as we address a bug at the lower level that triggers a crash on test exit.